utils/pattern/src/single.rs

```diff
@@ -153,7 +165,10 @@ pub enum SinglePlaceholder {}
 impl crate::private::Sealed for SinglePlaceholder {}
 
 impl PatternBackend for SinglePlaceholder {
-    type PlaceholderKey = SinglePlaceholderKey;
+    type PlaceholderKey<'a> = SinglePlaceholderKey;
+    #[cfg(feature = "alloc")]
+    type PlaceholderKeyCow<'a> = SinglePlaceholderKey;
```

Comment:
These are different types because one is allocky and requires serde whereas the other is fully no_std.

Reply:
Yeah it'd be nice if `Cow` was in `core` without the owned half.